File: backend/models/Vehicle.js
import mongoose from 'mongoose';

const vehicleSchema = new mongoose.Schema({
  userId: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
  name: { type: String, required: true },
  brand: { type: String, required: true },
  model: { type: String, required: true },
  year: { type: Number, required: true },
  batteryCapacity_kWh: { type: Number, required: true },
  currentCharge_percent: { type: Number, default: 50 },
  range_km: { type: Number, required: true },
  plateNumber: { type: String, default: '' },
  connectorType: { type: String, default: 'CCS' },
  color: { type: String, default: '#0ea5e9' },
  isDefault: { type: Boolean, default: false }
}, {
  timestamps: true
});

const Vehicle = mongoose.model('Vehicle', vehicleSchema);
export default Vehicle;

File: android/frontend/app/src/main/java/com/smartev/assistant/model/Vehicle.kt
package com.smartev.assistant.model

data class Vehicle(
    val id: String?,
    val _id: String?,
    val userId: String?,
    val name: String,
    val brand: String,
    val model: String,
    val year: Int,
    val batteryCapacity_kWh: Double,
    val currentCharge_percent: Int,
    val range_km: Int,
    val plateNumber: String,
    val isDefault: Boolean
)

data class VehicleResponse(
    val success: Boolean,
    val vehicles: List<Vehicle>?,
    val vehicle: Vehicle?,
    val message: String?
)
